diff --git a/quantum/rgb_matrix/rgb_matrix.c b/quantum/rgb_matrix/rgb_matrix.c
--- a/quantum/rgb_matrix/rgb_matrix.c
+++ b/quantum/rgb_matrix/rgb_matrix.c
@@ -154,11 +154,16 @@ __attribute__((weak)) int rgb_matrix_led_index(int index) {
     return index;
 }
 
+__attribute__((weak)) void rgb_matrix_color_observer_kb(uint8_t index, uint8_t red, uint8_t green, uint8_t blue) {}
+__attribute__((weak)) void rgb_matrix_color_all_observer_kb(uint8_t red, uint8_t green, uint8_t blue) {}
+
 void rgb_matrix_set_color(int index, uint8_t red, uint8_t green, uint8_t blue) {
+    rgb_matrix_color_observer_kb((uint8_t)index, red, green, blue);
     rgb_matrix_driver.set_color(rgb_matrix_led_index(index), red, green, blue);
 }
 
 void rgb_matrix_set_color_all(uint8_t red, uint8_t green, uint8_t blue) {
+    rgb_matrix_color_all_observer_kb(red, green, blue);
 #if defined(RGB_MATRIX_SPLIT)
     for (uint8_t i = 0; i < RGB_MATRIX_LED_COUNT; i++)
         rgb_matrix_set_color(i, red, green, blue);
diff --git a/quantum/rgb_matrix/rgb_matrix.h b/quantum/rgb_matrix/rgb_matrix.h
--- a/quantum/rgb_matrix/rgb_matrix.h
+++ b/quantum/rgb_matrix/rgb_matrix.h
@@ -153,5 +153,7 @@ int rgb_matrix_led_index(int index);
 
 void rgb_matrix_set_color(int index, uint8_t red, uint8_t green, uint8_t blue);
 void rgb_matrix_set_color_all(uint8_t red, uint8_t green, uint8_t blue);
+void rgb_matrix_color_observer_kb(uint8_t index, uint8_t red, uint8_t green, uint8_t blue);
+void rgb_matrix_color_all_observer_kb(uint8_t red, uint8_t green, uint8_t blue);
 
 void rgb_matrix_handle_key_event(uint8_t row, uint8_t col, bool pressed);
